langserve/server.py

```diff
@@ -47,10 +50,35 @@
     APIRouter = FastAPI = Any
 
 
-def _unpack_config(d: Union[BaseModel, Mapping], keys: Sequence[str]) -> Dict[str, Any]:
-    """Project the given keys from the given dict."""
-    _d = d.dict() if isinstance(d, BaseModel) else d
-    return {k: _d[k] for k in keys if k in _d}
+def _config_from_hash(config_hash: str) -> Dict[str, Any]:
+    try:
+        if not config_hash:
+            return {}
+
+        uncompressed = LZString.decompressFromEncodedURIComponent(config_hash)
+        parsed = json.loads(uncompressed)
+        if isinstance(parsed, dict):
+            return parsed
+        else:
+            raise HTTPException(400, "Invalid config hash")
+    except Exception:
+        raise HTTPException(400, "Invalid config hash")
+
+
+def _unpack_config(
+    *configs: Union[BaseModel, Mapping, str], keys: Sequence[str]
+) -> Dict[str, Any]:
+    """Merge configs, and project the given keys from the merged dict."""
+    config_dicts = []
+    for config in configs:
+        if isinstance(config, str):
+            config_dicts.append(_config_from_hash(config))
```

Comment:
Is it possible to squeeze this through a validator to make sure that the values also correspond to expected types?

Reply:
Done